src/main/java/org/openstreetmap/atlas/checks/validation/intersections/LineCrossingWaterBodyCheck.java

```diff
@@ -293,31 +302,61 @@ public boolean validCheckForObject(final AtlasObject object)
      *            {@link AtlasItem} crossing
      * @return whether given {@link AtlasItem} can cross a water body
      */
-    private boolean canCrossWaterBody(final AtlasItem crossingItem)
+    private boolean canCrossWaterBody(final AtlasItem crossingItem,
+            final GeometricSurface waterbody)
     {
-        // In the following cases, given item can cross a water body
-
-        return this.canCrossWaterBodyFilter.test(crossingItem)
+        // In the following cases, given line can cross a water body
+        if (this.canCrossWaterBodyFilter.test(crossingItem)
                 // It has a tag starting with addr
                 || crossingItem.containsKeyStartsWith(Collections.singleton(ADDRESS_PREFIX_KEY))
                 // If crossing item is a line and meets the conditions for a boundary
-                || crossingItem instanceof Line && isBoundary(crossingItem);
+                || crossingItem instanceof Line && isBoundary(crossingItem))
+        {
+            return true;
+        }
+
+        // Each intersection between the Edge & waterbody should have a Node in the atlas
+        if (!(crossingItem instanceof Edge)
+                || !this.intersectionsAreExplicit(waterbody, (Edge) crossingItem))
+        {
+            return false;
+        }
+
+        // All intersections between the street and waterbody
+        final Set<Location> intersections = waterbody instanceof Polygon
+                ? ((Polygon) waterbody).intersections(((Edge) crossingItem).asPolyLine())
+                : ((MultiPolygon) waterbody).outers().stream()
+                        .flatMap(polygon -> polygon
+                                .intersections(((Edge) crossingItem).asPolyLine()).stream())
+                        .collect(Collectors.toSet());
+        final Predicate<AtlasItem> hasFordTag = item -> !item.getTag(FordTag.KEY)
```

Comment:
Can the ford use case addition be added to the `lines.non_offending` configurable list?